### PR TITLE
Update PowerShell version

### PR DESCRIPTION
**Pull request by NRKirby**
Update the PowerShell version in the [Building, Debugging, and Testing on Windows](https://github.com/dotnet/roslyn/blob/e8f03e407226a43c122ce004c66e1795883f564b/docs/contributing/Building%2C%20Debugging%2C%20and%20Testing%20on%20Windows.md) docs to 5.0 as per issue #32241

### CODE REVIEW

File: eng/build.ps1 [hunk 1]
```diff
@@ -14,6 +14,12 @@
 # it's fine to call `build.ps1 -build -testDesktop` followed by repeated calls to
 # `.\build.ps1 -testDesktop`.
 
+if ($PSVersionTable.PSVersion.Major -lt "5") {
```
Comment (tmat) [inline eng/build.ps1]: I'd move it just before https://github.com/dotnet/roslyn/pull/33848/commits/947698d6b1bbe9b6597dd9421303968af6c7360e#diff-891b50b7d5fe7380fc246ed310019d61R69.

File: eng/build.ps1 [hunk 2]
```diff
@@ -14,6 +14,12 @@
...
+   $version = $PSVersionTable.PSVersion
```
Comment (tmat) [inline eng/build.ps1]: Inline this variable. `version $($PSVersionTable.PSVersion) detected`

File: eng/build.ps1 [hunk 3]
```diff
@@ -14,6 +14,12 @@
...
+   Write-Host "PowerShell version must be 5 or greater (version $version detected)"
+               exit 1
```
Comment (tmat) [inline eng/build.ps1]: To much indenting. `exit 1` should be indented the same as `Write-Host`.

Comment (NRKirby) [reply eng/build.ps1]: Done

File: eng/build.ps1 (hunk 1 above)
Comment (NRKirby) [reply eng/build.ps1]: Done

File: eng/build.ps1 (hunk 2 above)
Comment (NRKirby) [reply eng/build.ps1]: Done

**Comment by tmat** [comment]
@NRKirby thanks for the fix!

**Pull request merged**